backendv2/pkg/tofu/dl.go

```diff
@@ -0,0 +1,150 @@
+package tofu
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+)
+
+const (
+	nightliesBaseURL = "https://nightlies.opentofu.org"
+	latestURL        = nightliesBaseURL + "/nightlies/latest.json"
+)
+
+// TODO: Investigate if we should bring in tofudl here?
```

Comment:
Given that nightly support was merged, we should use it here + dogfood it.

Reply:
I know that we're actively maintaining tofudl for the language server as well, but I wasn't sure if we were recommending that it was the right way to download nightlies. 

This ties into a conversation I had with @diofeher thismorning. He made a small comment about nightlies and their support here: https://github.com/opentofu/opentofu/issues/3193#issuecomment-3467474908